internal/config/base.go

```diff
@@ -25,69 +25,97 @@ import (
 type Config interface {
 	ServiceHost() string
 	ServicePort() string
+	StripPrefix() string
 	LogLevel() string
 	LogFilePath() string
 	ETOSNamespace() string
 	DatabaseURI() string
+	PublicKey() ([]byte, error)
+	PrivateKey() ([]byte, error)
 }
 
-// cfg implements the Config interface.
-type cfg struct {
-	serviceHost   string
-	servicePort   string
-	logLevel      string
-	logFilePath   string
-	etosNamespace string
-	databaseHost  string
-	databasePort  string
+// baseCfg implements the Config interface.
+type baseCfg struct {
+	serviceHost    string
+	servicePort    string
+	stripPrefix    string
+	logLevel       string
+	logFilePath    string
+	etosNamespace  string
+	databaseHost   string
+	databasePort   string
+	publicKeyPath  string
+	privateKeyPath string
 }
 
-// Get creates a config interface based on input parameters or environment variables.
-func Get() Config {
-	var conf cfg
+// load the command line vars for a base configuration.
+func load() Config {
+	var conf baseCfg
 
 	flag.StringVar(&conf.serviceHost, "address", EnvOrDefault("SERVICE_HOST", "127.0.0.1"), "Address to serve API on")
 	flag.StringVar(&conf.servicePort, "port", EnvOrDefault("SERVICE_PORT", "8080"), "Port to serve API on")
+	flag.StringVar(&conf.stripPrefix, "stripprefix", EnvOrDefault("STRIP_PREFIX", ""), "Strip a URL prefix. Useful when a reverse proxy sets a subpath. I.e. reverse proxy sets /stream as prefix, making the etos API available at /stream/v1/events. In that case we want to set stripprefix to /stream")
 	flag.StringVar(&conf.logLevel, "loglevel", EnvOrDefault("LOGLEVEL", "INFO"), "Log level (TRACE, DEBUG, INFO, WARNING, ERROR, FATAL, PANIC).")
 	flag.StringVar(&conf.logFilePath, "logfilepath", os.Getenv("LOG_FILE_PATH"), "Path, including filename, for the log files to create.")
 	flag.StringVar(&conf.etosNamespace, "etosnamespace", ReadNamespaceOrEnv("ETOS_NAMESPACE"), "Path, including filename, for the log files to create.")
 	flag.StringVar(&conf.databaseHost, "databasehost", EnvOrDefault("ETOS_ETCD_HOST", "etcd-client"), "Host to the database.")
 	flag.StringVar(&conf.databasePort, "databaseport", EnvOrDefault("ETOS_ETCD_PORT", "2379"), "Port to the database.")
-
-	flag.Parse()
+	flag.StringVar(&conf.publicKeyPath, "publickeypath", os.Getenv("PUBLIC_KEY_PATH"), "Path to a public key to use for verifying JWTs.")
+	flag.StringVar(&conf.privateKeyPath, "privatekeypath", os.Getenv("PRIVATE_KEY_PATH"), "Path to a private key to use for signing JWTs.")
```

Comment:
Doesn't the private keys flag and config belong in the keys config?

Reply:
Hrm, on the other hand the private key is used by sse to validate the token.

Reply:
Correct. Public key is used in other places. Could move private key to only exist in the keys.go config file.